visualize-admin/visualization-tool
# perf: Fire DataCubesObservations query as soon as possible

Closes #1423

This PR removes a need to pass dimensions to get query filters, as it looks like this behavior is not needed. It was used to filter out interactive data filters that are a part of dimensions, which should always be true (except when a dimension was completely removed from a cube, which would most probably break the chart anyway).

With this change we fire `DataCubesObservations` query in parallel to `DataCubesComponents` fetching, not waiting for it to finish first.

## How to test
### PR
1. Go to [this link](https://visualization-tool-git-perf-fire-observations-query-asap-ixt1.vercel.app/en/v/ENrCApdv5Cbo?dataSource=Prod).
2. Open Developer Tools (e.g. [see how to do so on Chrome](https://developer.chrome.com/docs/devtools/open)).
3. Switch to the Network tab.
4. Reload the page.
5. ✅ See that the `graphql` queries were fired (almost) at the same time.

<img width="878" alt="Screenshot 2024-04-19 at 13 30 54" src="https://github.com/visualize-admin/visualization-tool/assets/52032047/dbf85c0c-3554-48fc-b925-e219fb81263a">

### TEST
1. Go to [this link](https://test.visualize.admin.ch/en/v/Zq6ql3-tONlE?dataSource=Prod).
2. Open Developer Tools (e.g. [see how to do so on Chrome](https://developer.chrome.com/docs/devtools/open)).
3. Switch to the Network tab.
4. Reload the page.
5. ❌ See that some graphql queries were fired sequentially (waited for other queries to finish first).

<img width="878" alt="Screenshot 2024-04-19 at 13 31 11" src="https://github.com/visualize-admin/visualization-tool/assets/52032047/a8038b71-8fbc-471a-8944-92aa49c74bb3">
